middleware.ml

```diff
@@ -40,105 +66,21 @@ let redirect_to_page ~path ?(clear_session = false) ?(with_error = false) reqd
         ]
       else []
     in
-    session_header
-    @ [
-        ("Set-Cookie", msg_cookie);
-        ("location", path);
-        ("Content-Length", string_of_int (String.length msg));
-      ]
+    session_header @ [ ("Set-Cookie", msg_cookie); ("location", path) ]
   in
-  let headers = H1.Headers.of_list header_list in
-  let response = H1.Response.create ~headers `Found in
-  H1.Reqd.respond_with_string reqd response msg;
-  Lwt.return_unit
-
-let redirect_to_error ~title ~data ~api_meth status reqd () =
-  let code = H1.Status.to_code status
-  and success = H1.Status.is_successful status in
-  let error = { Utils.Status.code; title; data; success } in
-  let data =
-    if api_meth then Utils.Status.to_json error
-    else
-      Guest_layout.guest_layout ~page_title:(title ^ " | Mollymawk")
-        ~content:(Error_page.error_layout error)
-        ~icon:"/images/robur.png" ()
-  in
-  Lwt.return
-    (let headers =
-       H1.Headers.of_list
-         [
-           ("content-length", string_of_int (String.length data));
-           ("content-type", if api_meth then "application/json" else "text/html");
-         ]
-     in
-     let resp = H1.Response.create ~headers status in
-     H1.Reqd.respond_with_string reqd resp data)
-
-let redirect_to_verify_email reqd ?(msg = "") () =
-  let headers =
-    H1.Headers.of_list
-      [
-        ("location", "/verify-email");
-        ("Content-Length", string_of_int (String.length msg));
-      ]
-  in
-  let response = H1.Response.create ~headers `Found in
-  H1.Reqd.respond_with_string reqd response msg;
-  Lwt.return_unit
-
-let redirect_to_url ~url reqd ?(msg = "") () =
-  let headers =
-    H1.Headers.of_list
-      [
-        ("location", url); ("Content-Length", string_of_int (String.length msg));
-      ]
-  in
-  let response = H1.Response.create ~headers `Found in
-  H1.Reqd.respond_with_string reqd response msg;
-  Lwt.return_unit
+  http_response ~api_meth:false ~title:"Redirecting" ~data:(`String "")
```

Comment:
``data:(`String "")`` -- we used to use `msg` here, which is now no longer used!?